### PR TITLE
Unit 13 - Outlook: I-JEPA draft

### PR DESCRIPTION
**Pull request by mmhamdy**
The chapter about I-JEPA from Unit 13: Outlook

Addressing issue #37

### CODE REVIEW

File: chapters/en/Unit 13 - Outlook/i-jepa.mdx [hunk 1]
```diff
@@ -0,0 +1,86 @@
+# Image-based Joint-Embedding Predictive Architecture (I-JEPA)
+
+## Overview
+
+The Image-based Joint-Embedding Predictive Architecture (I-JEPA) is a groundbreaking self-supervised learning model [introduced by Meta AI in 2023](https://ai.meta.com/blog/yann-lecun-ai-model-i-jepa/). It tackles the challenge of understanding images without relying on traditional labels or hand-crafted data augmentations.
+To get to know I-JEPA better, let’s first discuss a few concepts.
+
+### Invariance-based vs. Generative pretraining methods
```
Comment (merveenoyan) [inline chapters/en/Unit 13 - Outlook/i-jepa.mdx:8]:
```suggestion
### Invariance-based vs. Generative Pretraining Methods
```

File: chapters/en/Unit 13 - Outlook/i-jepa.mdx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+We can say that there are broadly two main approaches for self-supervised learning from images, Invariance-based methods and generative methods. Both approaches have their strengths and weaknesses.
```
Comment (merveenoyan) [inline chapters/en/Unit 13 - Outlook/i-jepa.mdx:10]:
```suggestion
We can say that there are broadly two main approaches for self-supervised learning from images: invariance-based methods and generative methods. Both approaches have their strengths and weaknesses.
```

File: chapters/en/Unit 13 - Outlook/i-jepa.mdx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+- **Invariance-based methods**: In these methods, the model tries to reproduce similar embeddings for different views of the same image. And of course, these different views are hand-crafted, the image augmentations that we’re all familiar with. For example: rotating, scaling, and cropping. These methods are good at producing representations at high semantic levels, but the problem is that they introduce strong biases that may be detrimental to certain downstream tasks. For example, image classification and instance segmentation do not require data augmentations.
```
Comment (merveenoyan) [inline chapters/en/Unit 13 - Outlook/i-jepa.mdx:12]:
```suggestion
- **Invariance-based methods**: In these methods, the model tries to reproduce similar embeddings for different views of the same image. And, of course, these different views are hand-crafted, the image augmentations we’re all familiar with. For example, rotating, scaling, and cropping. These methods are good at producing representations at high semantic levels, but the problem is that they introduce strong biases that may be detrimental to certain downstream tasks. For example, image classification and instance segmentation do not require data augmentations.
```

File: chapters/en/Unit 13 - Outlook/i-jepa.mdx [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+- **Generative methods**: In these methods, the model tries to reconstruct the input image. That’s why these methods are sometimes called reconstruction-based self-supervised learning. Masks are used to hide patches of the input image, and the model tries to reconstruct these corrupted patches either at the pixel or token level (let’s keep this point in mind). This masked approach can easily generalize beyond image modality but doesn’t produce representations at the quality level of invariance-based methods. Also, these methods are computationally expensive and require large datasets for robust training.
```
Comment (merveenoyan) [inline chapters/en/Unit 13 - Outlook/i-jepa.mdx:14]:
```suggestion
- **Generative methods**: The model tries to reconstruct the input image using these methods. That’s why these methods are sometimes called reconstruction-based self-supervised learning. Masks hide patches of the input image, and the model tries to reconstruct these corrupted patches at the pixel or token level (let’s keep this point in mind). This masked approach can easily generalize beyond image modality but doesn’t produce representations at the quality level of invariance-based methods. Also, these methods are computationally expensive and require large datasets for robust training.
```

File: chapters/en/Unit 13 - Outlook/i-jepa.mdx [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+
+Now let’s talk about Joint-Embedding Architecture.
+
+### Joint-Embedding Architectures
+
+This is a recent and popular approach for self-supervised learning from images in which two networks are trained to produce similar embeddings for different views of the same image. Basically, they train two networks to "speak the same language" about different views of the same picture. A common choice is the Siamese network architecture where the two networks share the same weights. But like everything else, it has its own problems:
+
+- **Representation collapse**: A case in which the model produces the same representation regardless of the input. Or in tech-speak, the energy landscape is flat.
```
Comment (merveenoyan) [inline chapters/en/Unit 13 - Outlook/i-jepa.mdx:22]:
```suggestion
- **Representation collapse**: A case in which the model produces the same representation regardless of the input.
```

File: chapters/en/Unit 13 - Outlook/i-jepa.mdx [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+
+2. **Context Encoder**: Encodes a randomly sampled context block from the image to obtain a corresponding patch-level representation.
+
+3. **Predictor**: Takes as input the output of the context encoder and a mask token for each patch we wish to predict and tries to predict the masked target blocks.
+
+The target-encoder, context-encoder, and predictor all use a Vision Transformer (ViT) architecture. You have a refresher about them in unit 3 of this course.
+
+The image below from the paper illustrates how I-JEPA works.
+
+![I-JEPA method](https://huggingface.co/datasets/hf-vision/course-assets/resolve/main/i-jepa-2.png?download=true)
```
Comment (merveenoyan) [inline chapters/en/Unit 13 - Outlook/i-jepa.mdx:62]:
```suggestion
![I-JEPA method](https://huggingface.co/datasets/hf-vision/course-assets/resolve/main/i-jepa-2.png)
```

File: chapters/en/Unit 13 - Outlook/i-jepa.mdx [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+
+## Why It Matters
+
+So, why I-JEPA? I-JEPA introduced many new design features while still being a simple and efficient method for learning semantic image representations without relying on hand-crafted data augmentations. In brief:
```
Comment (merveenoyan) [inline chapters/en/Unit 13 - Outlook/i-jepa.mdx:66]:
```suggestion
So, why I-JEPA? I-JEPA introduced many new design features while still being a simple and efficient method for learning semantic image representations without relying on hand-crafted data augmentations. Briefly,
```

**Review by merveenoyan — APPROVED** [review]
Mostly did nitpicking, otherwise LGTM! thank you!

File: chapters/en/Unit 13 - Outlook/i-jepa.mdx [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+
+- **Generative methods**: The model tries to reconstruct the input image using these methods. That’s why these methods are sometimes called reconstruction-based self-supervised learning. Masks hide patches of the input image, and the model tries to reconstruct these corrupted patches at the pixel or token level (let’s keep this point in mind). This masked approach can easily generalize beyond image modality but doesn’t produce representations at the quality level of invariance-based methods. Also, these methods are computationally expensive and require large datasets for robust training.
+
+Now let’s talk about Joint-Embedding Architecture.
```
Comment (johko) [inline chapters/en/Unit 13 - Outlook/i-jepa.mdx:16]: I think you should either add an 's' at the end or a 'the" before 'Joint-Embedding'
```suggestion
Now let’s talk about Joint-Embedding Architectures.
```

File: chapters/en/Unit 13 - Outlook/i-jepa.mdx [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+
+2. **Multi-block masking**: Another design choice that improves the semantic features produced by I-JEPA is masking sufficiently large blocks of the input image.
+
+### I-JEPA Components
+
+The previous diagrams show and compare the I-JEPA architecture, below is a brief description of its main components:
+
+1. **Target Encoder**: Encodes target images and the target blocks are produced by masking its output.
+
+2. **Context Encoder**: Encodes a randomly sampled context block from the image to obtain a corresponding patch-level representation.
```
Comment (johko) [inline chapters/en/Unit 13 - Outlook/i-jepa.mdx:54]: I think you should mention that 'y' has the target encoder and 'x' the context encoder attached when referring to the above image, as only the image after this part uses these words specifically

File: chapters/en/Unit 13 - Outlook/i-jepa.mdx [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+
+So, why I-JEPA? I-JEPA introduced many new design features while still being a simple and efficient method for learning semantic image representations without relying on hand-crafted data augmentations. Briefly,
+
+1. I-JEPA outperforms pixel-reconstruction methods such as Masked autoencoders (MAE) on ImageNet-1K linear probing, semi-supervised 1% ImageNet-1K, and semantic transfer tasks.
+
+2. I-JEPA is competitive with view-invariant pretraining approaches on semantic tasks and achieves better performance on low-level vision tasks such as object counting and depth prediction.
+
+3. By using a simpler model with less rigid inductive bias, I-JEPA is applicable to a wider set of tasks.
+
+4. I-JEPA is also scalable and efficient. Pretraining on ImageNet requires *less than 1200 GPU hours*.
```
Comment (johko) [inline chapters/en/Unit 13 - Outlook/i-jepa.mdx:74]: Any info on what kind of GPU they used for that? I feel like 1200 GPU hours on a A100 can be quite different to the same on a T4 (as an extreme example)

File: chapters/en/Unit 13 - Outlook/i-jepa.mdx (hunk 9 above)
Comment (mmhamdy) [reply chapters/en/Unit 13 - Outlook/i-jepa.mdx:54]: You're right, that makes more sense. Will add 'x' and 'y' references to the context and target encoders.

File: chapters/en/Unit 13 - Outlook/i-jepa.mdx (hunk 10 above)
Comment (mmhamdy) [reply chapters/en/Unit 13 - Outlook/i-jepa.mdx:74]: Yeah, they used 16 A100 GPUs for 72 hours.

**Pull request merged**